batchflow/tests/classification_metrics_test.py

```diff
@@ -0,0 +1,175 @@
+    conf_matrix = np.array([[2, 3], [1, 0]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='labels', num_classes=2)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+def test_confusion_matrix_multiclass():
+    """Test on correctness of confusion matrix calculation for multiclass"""
+    y_true, y_pred = np.array([2, 1]), np.array([[0.1, 0.1, 0.8], [0.1, 0.8, 0.1]])
+    conf_matrix = np.array([[0, 0, 0], [0, 1, 0], [0, 0, 1]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='proba', axis=1, num_classes=3)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+# tests on shapes
+
+@pytest.fixture(params=['accuracy', 'f1_score', 'true_positive_rate', 'false_positive_rate'])
+def metrics_name(request):
+    return request.param
+
+def test_diff_shapes_two_classes(metrics_name):
+    """Testing different shape"""
+    y_true_1, y_pred_1 = np.array([0, 1]), np.array(1)
+    y_true_2, y_pred_2 = np.array([[0, 1], [1, 0]]), np.array([0, 1])
+    y_true_3, y_pred_3 = np.array([[[0, 1], [1, 0]], [[1, 1], [0, 0]]]), np.array([[0, 1], [1, 0]])
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_1, y_pred_1, fmt='labels', num_classes=2), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_2, y_pred_2, fmt='labels', num_classes=2), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_3, y_pred_3, fmt='labels', num_classes=2), metrics_name)()
+
+def test_diff_shapes_multiclass(metrics_name):
+    """Testing different shapes for multiclass"""
+    y_true_1, y_pred_1 = np.array(2), np.array([[0.8, 0.1, 0.1], [0.1, 0.1, 0.8]])
+    y_true_2 = np.array([[[0, 1, 0], [0, 1, 0], [1, 0, 0], [0, 0, 1]], [[1, 0, 0], [1, 1, 0], [1, 0, 1], [0, 0, 1]]])
+    y_pred_2 = np.array([[0.1, 0.8, 0.1], [0.1, 0.8, 0.1], [0.8, 0.1, 0.1], [0.8, 0.1, 0.1]])
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_1, y_pred_1, fmt='proba', axis=1, num_classes=3), metrics_name)()
+    with pytest.raises(ValueError):
+        getattr(cm(y_true_2, y_pred_2, fmt='proba', axis=1, num_classes=3), metrics_name)()
+
+@pytest.mark.parametrize('metrics, metrics_shape', [
+    ('accuracy', ()),
+    ('f1_score', (1, 1)),
+    ('true_positive_rate', (1, 1)),
+    ('false_positive_rate', (1, 1))
+])
+def test_single_value_two_class(metrics, metrics_shape):
+    """Test on various metrics: output single value in case of two class classification"""
+    y_true, y_pred = np.random.choice([0, 1], size=(5,)), np.random.choice([0, 1], size=(5,))
+    test_shape = getattr(cm(y_true, y_pred, fmt='labels', num_classes=2), metrics)().shape
+    assert test_shape == metrics_shape
+
+@pytest.mark.parametrize('metrics, metrics_shape', [
+    ('accuracy', ()),
+    ('f1_score', (1, 1)),
+    ('true_positive_rate', (1, 1)),
+    ('false_positive_rate', (1, 1))
+])
```

Comment:
Trick with `.reshape` is applicable here as well (and for every test for shape below)